src/main/fc/rc.c

```diff
@@ -641,57 +550,55 @@ FAST_CODE void processRcCommand(void)
         checkForThrottleErrorResetState(currentRxRefreshRate);
     }
 
-#ifdef USE_INTERPOLATED_SP
     if (isRxDataNew) {
-        for (int i = FD_ROLL; i <= FD_YAW; i++) {
-            isDuplicate[i] = (oldRcCommand[i] == rcCommand[i]);
-            rcCommandDelta[i] = fabsf(rcCommand[i] - oldRcCommand[i]);
-            oldRcCommand[i] = rcCommand[i];
-            float rcCommandf;
-            if (i == FD_YAW) {
-                rcCommandf = rcCommand[i] / rcCommandYawDivider;
-            } else {
-                rcCommandf = rcCommand[i] / rcCommandDivider;
-            }
-            const float rcCommandfAbs = fabsf(rcCommandf);
-            rawSetpoint[i] = applyRates(i, rcCommandf, rcCommandfAbs);
-        }
-    }
-#endif
+        for (int axis = FD_ROLL; axis <= FD_YAW; axis++) {
 
-    switch (rxConfig()->rc_smoothing_type) {
-#ifdef USE_RC_SMOOTHING_FILTER
-    case RC_SMOOTHING_TYPE_FILTER:
-        updatedChannel = processRcSmoothingFilter();
-        break;
-#endif // USE_RC_SMOOTHING_FILTER
-    case RC_SMOOTHING_TYPE_INTERPOLATION:
-    default:
-        updatedChannel = processRcInterpolation();
-        break;
-    }
+            isDuplicate[axis] = (oldRcCommand[axis] == rcCommand[axis]);
+            rcCommandDelta[axis] = fabsf(rcCommand[axis] - oldRcCommand[axis]);
+            oldRcCommand[axis] = rcCommand[axis];
 
-    if (isRxDataNew || updatedChannel) {
-        const uint8_t maxUpdatedAxis = isRxDataNew ? FD_YAW : MIN(updatedChannel, FD_YAW); // throttle channel doesn't require rate calculation
-#if defined(SIMULATOR_BUILD)
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wunsafe-loop-optimizations"
-#endif
-        for (int axis = FD_ROLL; axis <= maxUpdatedAxis; axis++) {
-#if defined(SIMULATOR_BUILD)
-#pragma GCC diagnostic pop
+            float angleRate;
+            
+#ifdef USE_GPS_RESCUE
```

Comment:
Quick question, has GPS rescue been tested after this change?

Reply:
No, it hasn't.

Reply:
We should probably keep this in mind and make sure this aspect gets tested before this goes into a release.

Reply:
Yes, I have asked for people to check this now.